### PR TITLE
Allow camel component to be initialized

### PR DESCRIPTION
**Pull request by ctron**
Without OSGi blueprint support for Camel some components cannot be
properly initialized. Therefore the CloudServiceFactory allows to set
a init code snippet for setting up Camel components before starting
the router.

The same change is needed for the Kura apps based Camel router. But this
change has to wait until PR #756 is merged.

Signed-off-by: Jens Reimann <jreimann@redhat.com>

### CODE REVIEW

**Comment by ctron** [comment]
So in order to test this:
* Install the DP: https://repo1.maven.org/maven2/de/dentrassi/kura/addons/de.dentrassi.kura.addons.amqp/0.1.0/de.dentrassi.kura.addons.amqp-0.1.0.dp
* Create a new Camel cloud service
* Edit it and add the following configuration:

XML:
```xml
<routes xmlns="http://camel.apache.org/schema/spring">
<route id="foo">
<from uri="timer:foo"/>
<setBody><constant>foo</constant></setBody>
<to uri="my-amqp:topic:foobar"/>
</route>
</routes>
```

JavaScript:
```js
var amqp = org.apache.camel.component.amqp.AMQPComponent.amqpComponent("amqp://localhost:5672");
camelContext.addComponent ( "my-amqp", amqp );
```

You should see connection issues in the log complaining about a connection to `localhost:5672`, since you don't have a broker running.

If the JS snippet would not work you would have errors about `my-amqp` not found.

File: kura/org.eclipse.kura.camel.test/src/org/eclipse/kura/camel/runner/ScriptRunnerTest.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Red Hat Inc and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat Inc - initial API and implementation
```
Comment (MMaiero) [inline kura/org.eclipse.kura.camel.test/src/org/eclipse/kura/camel/runner/ScriptRunnerTest.java]: Please update all the copyright headers to the new format

**Comment by ctron** [comment]
Done

**Comment by MMaiero** [comment]
For all the files? I still see the old in XmlCamelCloudService.java and the other files. Thanks

**Comment by ctron** [comment]
We can do this later. I did it for all the files that got touched by this PR.

If you think it is that important to hold up the PR then fine.

**Comment by MMaiero** [comment]
ok for me

**Comment by ctron** [comment]
So ready to merge?

**Pull request merged**